packages/integrations/tailwind/src/index.ts

```diff
@@ -28,42 +33,49 @@ async function getUserConfig(root: URL, configPath?: string, isRestart = false)
 		userConfigPath = fileURLToPath(new URL(configPathWithLeadingSlash, root));
 	}
 
+	let resolvedConfigPath = '';
+	if (!configPath) {
+		if (path.join(resolvedRoot, 'tailwind.config.js')) {
+			resolvedConfigPath = path.join(resolvedRoot + 'tailwind.config.js');
+		} else if (existsSync(path.join(resolvedRoot + 'tailwind.config.ts'))) {
+			resolvedConfigPath = path.join(resolvedRoot + 'tailwind.config.ts');
+		} else if (existsSync(path.join(resolvedRoot + 'tailwind.config.cjs'))) {
+			resolvedConfigPath = path.join(resolvedRoot + 'tailwind.config.cjs');
+		} else if (existsSync(path.join(resolvedRoot + 'tailwind.config.mjs'))) {
+			resolvedConfigPath = path.join(resolvedRoot + 'tailwind.config.mjs');
+		} else {
+			throw new Error('No tailwind config found at project root');
+		}
+	}
+
+	const configPathToUse = userConfigPath ?? resolvedConfigPath;
+
+	if (!loadConfig && configPathToUse.endsWith('ts')) {
+		throw new Error(
+			'To use a typescript tailwind config file please install tailwind 3.3.0 or higher'
+		);
+	}
+
+	// previous to tailwindcss 3.3.0 loadConfig did not exist
+
 	if (isRestart) {
-		// Hack: Write config to temporary file at project root
-		// This invalidates and reloads file contents when using ESM imports or "resolve"
-		const resolvedConfigPath = (await resolve('tailwind', {
-			mustExist: false,
-			cwd: resolvedRoot,
-			filePath: userConfigPath,
-		})) as string;
-
-		const { dir, base } = path.parse(resolvedConfigPath);
-		const tempConfigPath = path.join(dir, `.temp.${Date.now()}.${base}`);
-		await fs.copyFile(resolvedConfigPath, tempConfigPath);
-
-		let result: load.Config<Record<any, any>> | undefined;
+		clearModule(configPathToUse);
+		if (configPathToUse.endsWith('ts')) {
+			return { config: loadConfig(configPathToUse), configPath: configPathToUse };
+		} else {
+			return { config: (await import(configPathToUse))?.default, configPath: configPathToUse };
+		}
+	} else {
 		try {
-			result = await load('tailwind', {
-				mustExist: false,
-				cwd: resolvedRoot,
-				filePath: tempConfigPath,
-			});
+			if (configPathToUse.endsWith('ts')) {
```

Comment:
```suggestion
			if (configPathToUse.endsWith('ts') || configPathToUse.endsWith('mjs')) {
```

prob time to pull this out to an `isESM` variable now xdd